### PR TITLE
Feature/url queries

### PR DESCRIPTION
**Pull request by jordaniza**
Link the filters panel to the router and create deeplink functionality to load the filters from the URL on first load.

### CODE REVIEW

File: packages/react-app/src/components/pages/Bounties/index.tsx [hunk 1]
```diff
@@ -28,45 +31,99 @@ const FilterResultPlaceholder = ({ message }: { message: string }): JSX.Element
...
+};
+
+const getFiltersFromUrl = (query: ParsedUrlQuery): FilterParams => Object.entries(query).reduce((prev, [key, val]) => {
+	/**
+	 * Grab filters from the url, using fallback values if we see 'undefined'
+	 */
+	const isValid = (val && val !== 'undefined');
+	const existing = baseFilters[key as keyof FilterParams];
+	const adjVal = isValid ? val : existing;
+	console.debug({ key, val, existing, adjVal, isValid });
```
Comment (Behold3th) [inline packages/react-app/src/components/pages/Bounties/index.tsx:84]: debug log

File: packages/react-app/src/components/pages/Bounties/index.tsx [hunk 2]
```diff
@@ -28,45 +31,99 @@ const FilterResultPlaceholder = ({ message }: { message: string }): JSX.Element
...
+
+	}, [filters, BANKLESS, customer, debounceSearch]);
+};
+
+const usePaginatedBounties = (bounties: BountyCollection[] | undefined, page: number) => {
+	// splits bounties according to the maximum page size
+	return useMemo(() => bounties
+		? bounties.slice(PAGE_SIZE * page, Math.min(bounties.length, PAGE_SIZE * (page + 1)))
+		: []
+	, [bounties, page, PAGE_SIZE]);
```
Comment (Behold3th) [inline packages/react-app/src/components/pages/Bounties/index.tsx:74]: why is `bounties` one of the dependencies for `useMemo`?

Comment (Behold3th) [inline packages/react-app/src/components/pages/Bounties/index.tsx:74]: why is `bounties` one of the dependencies for `useMemo`?

File: packages/react-app/src/types/Filter.ts [hunk 3]
```diff
@@ -6,17 +6,21 @@ export type AcceptedSortInputs = 'reward' | 'createdAt';
...
-	status?: string;
+	status?: string
 	search: string;
-	$lte?: number;
-	$gte?: number;
+	lte?: number;
+	gte?: number;
 	sortBy?: string;
-	asc: boolean;
+	asc: boolean | string;
```
Comment (Behold3th) [inline packages/react-app/src/types/Filter.ts:14]: I thought the creation/use of `booleanAsc` method was spot on, which contradicts adding `string` as a supported type for `asc`

Comment (Behold3th) [inline packages/react-app/src/types/Filter.ts:14]: I thought the creation/use of `booleanAsc` method was spot on, which contradicts adding `string` as a supported type for `asc`

File: packages/react-app/src/components/pages/Bounties/index.tsx [hunk 4]
```diff
@@ -28,45 +31,99 @@ const FilterResultPlaceholder = ({ message }: { message: string }): JSX.Element
 		justify="center"
 		align="center"
 	>
-		<Text	fontSize="lg">{ message }</Text>
+		<Text fontSize="lg">{message}</Text>
 	</Stack>
 );
 
+const baseFilters: FilterParams = {
```
Comment (Behold3th) [inline packages/react-app/src/components/pages/Bounties/index.tsx:38]: I think these are very sensible defaults, nice

Comment (Behold3th) [inline packages/react-app/src/components/pages/Bounties/index.tsx:38]: I think these are very sensible defaults, nice

File: packages/react-app/src/components/pages/Bounties/Filters/index.tsx [hunk 5]
```diff
@@ -147,18 +163,8 @@ const MinMaxFilter = ({ name, setLte, setGte }: {
 };
 
 const Filters = (props: {
-	status: string,
```
Comment (Behold3th) [inline packages/react-app/src/components/pages/Bounties/Filters/index.tsx:150]: Nice!

Comment (Behold3th) [inline packages/react-app/src/components/pages/Bounties/Filters/index.tsx:150]: Nice!

File: packages/react-app/src/components/pages/Bounties/Filters/index.tsx [hunk 6]
```diff
@@ -121,18 +135,20 @@ const SortBy = ({ name, options, sortBy, sortAscending, setSortBy, setSortAscend
...
-	lte: number,
-	setLte: SetState<number>,
-	gte: number,
-	setGte: SetState<number>
-}): JSX.Element => {
-	const updateMin = (event: any): void => {
-		setGte(event.target.value);
+} & UseFilterState): JSX.Element => {
+	const updateMin = (event: Event): void => {
+		setFilters({
```
Comment (Behold3th) [inline packages/react-app/src/components/pages/Bounties/Filters/index.tsx:142]: This new pattern is great, really clean

Comment (Behold3th) [inline packages/react-app/src/components/pages/Bounties/Filters/index.tsx:142]: This new pattern is great, really clean

**Review by Behold3th — DISMISSED** [review]
Exciting ! 💯

File: packages/react-app/src/components/pages/Bounties/index.tsx (hunk 1 above)
Comment (jordaniza) [reply packages/react-app/src/components/pages/Bounties/index.tsx:84]: Will remove

File: packages/react-app/src/components/pages/Bounties/index.tsx (hunk 2 above)
Comment (jordaniza) [reply packages/react-app/src/components/pages/Bounties/index.tsx:74]: Best practice with react is to use exhaustive dependencies with useMemo, useCallback and useEffect.

https://stackoverflow.com/questions/58866796/understanding-the-react-hooks-exhaustive-deps-lint-rule

File: packages/react-app/src/types/Filter.ts (hunk 3 above)
Comment (jordaniza) [reply packages/react-app/src/types/Filter.ts:14]: asc is a query string param, so is parsed from the url as the string `true` or `false`. Maybe I can address this.

File: packages/react-app/src/components/pages/Bounties/Bounty/index.tsx [hunk 7]
```diff
@@ -157,7 +157,7 @@ export const AccordionBountyItem = ({ bounty }: { bounty: BountyCollection }): J
 	<AccordionItem borderWidth={3} borderRadius={10} mb={3}>
 		<AccordionButton pb={5}>
 			<BountySummary bounty={bounty} />
-			<Box pos="relative" textAlign="right" w={0} left={-4} top={-7}>
+			<Box pos="relative" textAlign="right" w={0} left={-4} top={{ base: -10, md: -5 }}>
```
Comment (jordaniza) [inline packages/react-app/src/components/pages/Bounties/Bounty/index.tsx:160]: ok 2 line change

**Comment by jordaniza** [comment]
Merging due to no further comments

**Pull request merged**